libs/sdk-bindings/src/breez_sdk.udl

```diff
@@ -628,10 +649,10 @@ interface BlockingBreezServices {
    BuyBitcoinResponse buy_bitcoin(BuyBitcoinRequest req);
 };
 
-namespace breez_sdk {  
+namespace breez_sdk {
 
  [Throws=SdkError]
- BlockingBreezServices connect(Config config, sequence<u8> seed, EventListener listener);
+ BlockingBreezServices connect(Config config, sequence<u8> seed, EventListener listener, Logger node_logger, string? log_file_path);
```

Comment:
I think it's time to wrap this into a `ConnectRequest`. 
```
dictionary ConnectRequest {
    Config config;
    sequence<u8> seed;
    EventListener listener;
    Logger? logger=null;
    string? log_file_path=null;
};
```

The logger can probably be optional. I think by default we can initialize our own logger that logs to stdout. (will that work? I don't know). If the client sets a logger, we use that instead.

Reply:
Interesting! I do agree (and it will help for the multi-instances static_backup static call as well)!

I will give it a try!

Reply:
I quickly tried to implement the "connectRequest" yesterday and it's not trivial. The problem is that you can't simply put a trait in an udl structure (because you have to implement Equalable, Hashable,...). So we should rather work with references, but after trying, I don't get the impression that it works... (I've followed their documentation, but no success... It was a quick try but it still takes a bit of work to try it out maybe)

In any case, if we do that, we'll have to change a lot of things, (add Arc everywhere, even for the evenHandler and then we'll have pretty things like:
"self.event_listener.as_ref().as_ref().unwrap().on_event(e.clone())")

Given the changes this would involve, I wonder if this is still within the scope of the PR? or should it be a separated PR?

If you think so, I'll look into it seriously!

Reply:
> I quickly tried to implement the "connectRequest" yesterday and it's not trivial. The problem is that you can't simply put a trait in an udl structure (because you have to implement Equalable, Hashable,...). So we should rather work with references, but after trying, I don't get the impression that it works... (I've followed their documentation, but no success... It was a quick try but it still takes a bit of work to try it out maybe)
> 
> In any case, if we do that, we'll have to change a lot of things, (add Arc everywhere, even for the evenHandler and then we'll have pretty things like: "self.event_listener.as_ref().as_ref().unwrap().on_event(e.clone())")
> 
> Given the changes this would involve, I wonder if this is still within the scope of the PR? or should it be a separated PR?
> 
> If you think so, I'll look into it seriously!

@Torakushi given this analysis it is definitely not part of this PR which is already large enough.